static/app/components/profiling/flamegraph/flamegraphDrawer/flamegraphTreeTable.tsx

```diff
@@ -176,6 +177,42 @@ export function FlamegraphTreeTable({
     [contextMenu, formatDuration, referenceNode, getFrameColor]
   );
 
+  // This is slighlty unfortunate and ugly, but because our two columns are sticky
+  // we need to scroll the container to the left when we scroll to a node. This
+  // should be resolved when we split the virtualization between containers and sync scroll,
+  // but is a larger undertaking and will take a bit longer
+  const onScrollToNode: UseVirtualizedTreeProps<FlamegraphFrame>['onScrollToNode'] =
+    useCallback(
+      (
+        node: VirtualizedTreeRenderedRow<FlamegraphFrame> | undefined,
+        scrollContainer: HTMLElement | null,
+        coordinates?: {depth: number; top: number}
+      ) => {
+        if (node) {
+          const lastCell = node.ref?.lastChild?.firstChild as
+            | HTMLElement
+            | null
+            | undefined;
+          if (lastCell) {
+            lastCell.scrollIntoView({
+              block: 'nearest',
+            });
+
+            const left = -328 + (node.item.depth * 14 + 8);
+            scrollContainer?.scrollBy({
+              left,
+            });
+          }
+        } else if (coordinates) {
```

Comment:
There are basically two modes of how scrollTo can be used, one is when it is fired after a user key press in which case the next node is already rendered and we can give the user the reference to the DOM node so you can call scrollIntoView and get the "nearest" behavior for free w/o having to compute it yourself.

The other mode is when the scroll is fired programmatically, when that happens the tree will scroll to a position where the node may not yet have been rendered (e.g jumping half the view). When that happens, we give user the precomputed coordinates of that node (rowHeight * index and node depth) so they can scroll to the right position in the container